src/include/txn_inline.h

```diff
@@ -1089,13 +1084,11 @@ __wt_txn_read(
         have_stop_tw = WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw);
 
         /* Check the ondisk value. */
-        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          &is_ovfl_rm
-#endif
-        );
-        WT_RET(ret);
+        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &tw);
+        if (ret == WT_RESTART)
+            is_ovfl_rm = true;
```

Comment:
There are a bunch of things that are executed before we make use of this variable, is it ok?

Reply:
It is intentional. We should only retry if the onpage value is the one we want to return. Otherwise, no need to retry read.